src/lib/Components/Bidding/Screens/__tests__/ConfirmBid-tests.tsx

```diff
@@ -598,22 +640,16 @@ describe("ConfirmBid for unqualified user", () => {
       </BiddingThemeProvider>
     )
 
-    // manually setting state to avoid duplicating tests for UI interaction, but practically better not to do so.
-    component.root.instance.setState({ billingAddress })
-    component.root.instance.setState({ creditCardToken: stripeToken })
-    component.root.findByType(Checkbox).instance.props.onPress()
-    component.root.findByType(Button).instance.props.onPress()
+    fillOutFormAndSubmit(component)
 
-    jest.runAllTicks()
-    expect(nextStep.component).toEqual(BidResultScreen)
-    expect(nextStep.passProps).toEqual(
-      expect.objectContaining({
-        bidderPositionResult: {
-          message_header: "An error occurred",
-          message_description_md: "Your bid couldn’t be placed. Please\ncheck your internet connection\nand try again.",
-        },
-      })
+    expect(stripe.createTokenWithCard.mock.calls.length).toEqual(1)
+
+    const modal = component.root.findByType(Modal)
+
+    expect(modal.props.detailText).toEqual(
+      "There was a problem processing your information. Check your payment details and try again."
     )
+    expect(modal.props.visible).toEqual(true)
```

Comment:
It seems like now we have a modal within the `<ConfirmBid>` component to show errors, so updating the old test to reflect the actual change.